examples/matrice-factorisation/Main.hs

```diff
@@ -0,0 +1,65 @@
+{-# LANGUAGE BlockArguments #-}
+{-# LANGUAGE FunctionalDependencies #-}
+{-# LANGUAGE RecordWildCards #-}
+
+module Main where
+
+import Control.Monad (foldM, when)
+import Torch.Autograd (grad, makeIndependent, toDependent)
+import Torch.DType (DType (Float))
+import Torch.Functional (matmul, mse_loss)
+import Torch.NN
+  ( Parameter,
+    Randomizable,
+    sample,
+    sgd,
+  )
+import Torch.Tensor (Tensor, asTensor)
+import Torch.TensorFactories (randn')
+
+data MatrixFactSpec = MatrixFactSpec {common_dim :: Int, extra_dim :: Int} deriving (Show, Eq)
+
+data MatrixFact = MatrixFact {u :: Parameter, v :: Parameter}
+
+instance Randomizable MatrixFactSpec MatrixFact where
+  sample (MatrixFactSpec n k) = do
+    u <- makeIndependent =<< randn' [n, k]
+    v <- makeIndependent =<< randn' [k, n]
+    pure $ MatrixFact u v
+
+flattenParameters :: MatrixFact -> [Parameter]
+flattenParameters (MatrixFact u v) = [u, v]
+
+replaceParameters :: MatrixFact -> [Parameter] -> MatrixFact
+replaceParameters state params_list = MatrixFact u v
+  where
+    u = params_list !! 0
+    v = params_list !! 1
```

Comment:
it *should* be possible to use `Parameterized` from `Torch.Parameter`

Reply:
@tscholak 
I am not sure yet how to do it. Will look at it and come back to you.

Reply:
I have not succeeded to write `MatrixFact` as an instance of `Parameterized`.

Reply:
ok, never mind, you did an amazing job nonetheless 😃

Reply:
Thanks!